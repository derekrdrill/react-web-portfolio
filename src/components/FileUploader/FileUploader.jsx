import React, { useState } from 'react';
import axios from 'axios';

export const FileUploader = () => {
  const [files, setFiles] = useState([]);

  const onChange = e => {
    setFiles(e.target.files);
  };

  const onSubmit = async () => {
    // e.preventDefault();
    const formData = new FormData();
    Object.values(files).forEach(file => {
      console.log(file);
      formData.append('uploadImages', file);
    });

    console.log(formData);

    // try {
    //   //   const res =
    //   await axios.post('../../upload-images', formData, {
    //     headers: {
    //       'Content-Type': 'multipart/form-data',
    //     },
    //   });
    //   //   console.log(res);
    // } catch (err) {
    //   if (err.response.status === 500) {
    //     console.log(err);
    //   } else {
    //     console.log(err.response.data.msg);
    //   }
    // }
  };

  return (
    <div>
      <input type='file' multiple onChange={onChange} />
      <br />
      <br />
      <button onClick={onSubmit}>Submit</button>
    </div>
  );
};
