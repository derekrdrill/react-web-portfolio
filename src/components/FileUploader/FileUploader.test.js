import 'regenerator-runtime/runtime';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { FileUploader } from './FileUploader';

const renderer = new ShallowRenderer();

describe('File Uploader View tests', () => {
  it('renders correctly', () => {
    renderer.render(<FileUploader open={true} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
