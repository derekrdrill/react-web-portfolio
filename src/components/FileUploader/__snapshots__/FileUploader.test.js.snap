// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Uploader View tests renders correctly 1`] = `
<div>
  <input
    multiple={true}
    onChange={[Function]}
    type="file"
  />
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
